app/worker/http.go

```diff
@@ -39,33 +40,42 @@ var makeHTTPRequest = func(req *http.Request, timeout time.Duration) (*http.Resp
 	return res, nil
 }
 
-func processHTTPTask(httpTask *model.HTTPTask, job *model.Job,
-	workflow *model.Workflow, l *log.Logger) (*model.TaskResult, error) {
-	uri := processJobString(httpTask.URI, workflow, job)
+func processHTTPTask(
+	httpTask *model.HTTPTask,
+	ps *processor.JobStringProcessor,
+	l *log.Logger,
+) (*model.TaskResult, error) {
+	options := &processor.Options{Encoding: processor.URL}
+	uri := ps.ProcessJobString(httpTask.URI, options)
+	l.Infof("processHTTPTask: starting with: method=%s uri=%s (options=%+v)",
+		httpTask.Method,
+		uri,
+		options,
+	)
 
 	var payloadString string
 	if len(httpTask.FormData) > 0 {
 		form := url.Values{}
 		for key, value := range httpTask.FormData {
-			key = processJobString(key, workflow, job)
-			key = maybeExecuteGoTemplate(key, job.InputParameters.Map())
-			value = processJobString(value, workflow, job)
-			value = maybeExecuteGoTemplate(value, job.InputParameters.Map())
+			key = ps.ProcessJobString(key, nil)
+			key = maybeExecuteGoTemplate(key, ps.GetJobInputParameters().Map())
```

Comment:
It looks weird obtaining the job's input parameters from a `GetJobInputParameters` method on the JobStringProcessor instance. It looks like you did it by convenience because processHTTPTask doesn't receive anymore the job, but honestly I don't like it too much.

Reply:
I know, but there is no good way of going it. would you prefer `ps.GetJob()` method instead?

Reply:
@alfrunes there is a bit of refactoring going on here (I could not stop) in order not to redesign the whole thing, I added `GetJobInputParameters` in the string processor class. we (Fabio and me) both think it is not very elegant, and a bit out of place. what do you think? (Keep in mind that we do not want to go over board here :>)

Reply:
Since `maybeExecuteGoTemplate` is called wherever `ps.ProcessJobString` is called; wouldn't it be easier to just move it inside the `processor` package?

Reply:
brilliant. so I created a method `MaybeExecuteGoTemplate` but there were complications, in the end, this is the final version. makes sense? beware: I can refactor deeper.